provider/awssd/aws_sd.go

```diff
@@ -183,6 +183,12 @@ func (p *AWSSDProvider) instancesToEndpoint(ns *sd.NamespaceSummary, srv *sd.Ser
 	}
 
 	for _, inst := range instances {
+		// Ignore any instances that don't match our expected owner label.
+		if inst.Attributes[endpoint.OwnerLabelKey] == nil || *inst.Attributes[endpoint.OwnerLabelKey] != *srv.Description {
+			log.Debugf("Skipping instance \"%v\" found in service \"%v\", not under management", *inst.Id, *srv.Name)
+			continue
+		}
+
```

Comment:
Ownership concept should be handled in `registry`, not directly in the `provider`. See `AWSSDRegistry`, it may be the right spot to do this filtering.